lib/piper/permissions/ast/chars/rule.ex

```diff
@@ -3,7 +3,12 @@ defimpl String.Chars, for: Piper.Permissions.Ast.Rule do
   alias Piper.Permissions.Ast
 
   def to_string(%Ast.Rule{command_selector: cs, permission_selector: ps}) do
-    "when #{cs} must have #{ps}"
+    "when #{cs} #{ps_str(ps)}"
   end
 
+  defp ps_str(%Ast.ConditionalExpr{op: :allow}=ps),
+    do: "#{ps}"
+  defp ps_str(ps),
+    do: "must have #{ps}"
```

Comment:
@kevsmith Before the update we always stuck the "must have" bit in before the permission selector since all permissions started with it, see the diff for line 6. Now that we have rules that look like `when command is foo:bar allow` we just want the "must have" bit when the permission isn't `allow`.

@vanstee that's actually what I thought first. But you can have multiple `ConditionalExpr` in the permission_selector. So if the `to_string` for `ConditionalExpr` inserted it you could end up with extra "must have's".